internal/handlers/pg/pgdb/collections.go

```diff
@@ -32,7 +32,8 @@ import (
 )
 
 // validateCollectionNameRe validates collection names.
-var validateCollectionNameRe = regexp.MustCompile("^[a-zA-Z_-][a-zA-Z0-9_-]{0,119}$")
+// Empty collection name, names with `$` and `\x00` are not allowed.
+var validateCollectionNameRe = regexp.MustCompile("^[^$\x00]{1,235}$")
```

Comment:
See https://www.mongodb.com/docs/manual/reference/limits/#mongodb-limit-Restriction-on-Collection-Names

> 235 bytes for sharded collections

This is where I got 235 from.

Reply:
Can we introduce a test case that checks for exactly 235 bytes? Or maybe update the one that contains 100 bytes.

Reply:
Good point, I've replaced the test with 235.